addon/serializers/json.js

```diff
@@ -829,15 +829,21 @@ export default Serializer.extend({
   /**
     Check if the given hasMany relationship should be serialized
 
-    @method _shouldSerializeHasMany
-    @private
+    @method shouldSerializeHasMany
     @param {DS.Snapshot} snapshot
     @param {String} key
     @param {String} relationshipType
     @return {boolean} true if the hasMany relationship should be serialized
   */
-  _shouldSerializeHasMany(snapshot, key, relationship) {
-    var relationshipType = snapshot.type.determineRelationshipType(relationship, this.store);
+  shouldSerializeHasMany(snapshot, key, relationship) {
+    if (this._shouldSerializeHasMany) {
+      deprecate('The private method `_shouldSerializeHasMany` has been deprecated. Please use public method `shouldSerializeHasMany`.', false, {
```

Comment:
I think "The private method `_shouldSerializeHasMany` has been promoted to the public API. Please remove the underscore to use the public `shouldSerializeHasMany` method instead."